sdk/keyvault/azkeys/utils_test.go

```diff
@@ -40,48 +61,37 @@ func TestMain(m *testing.M) {
 			panic(err)
 		}
 	case recording.RecordingMode:
-		vaultUrl := os.Getenv("AZURE_KEYVAULT_URL")
-		err := recording.AddURISanitizer(fakeKvURL, vaultUrl, nil)
-		if err != nil {
-			panic(err)
-		}
-
-		err = recording.AddBodyKeySanitizer("$.key.kid", fakeKvURL, vaultUrl, nil)
-		if err != nil {
-			panic(err)
-		}
-
-		err = recording.AddBodyKeySanitizer("$.recoveryId", fakeKvURL, vaultUrl, nil)
-		if err != nil {
-			panic(err)
+		if liveVaultURL == "" {
+			panic("no value for AZURE_KEYVAULT_URL")
 		}
-
-		tenantID := os.Getenv("AZKEYS_TENANT_ID")
-		err = recording.AddHeaderRegexSanitizer("WWW-Authenticate", "00000000-0000-0000-0000-000000000000", tenantID, nil)
```

Comment:
Why was this removed?

Reply:
There's no need to replace tenant IDs because they aren't secret or considered in request matching. However, when the variable isn't set and `tenantID == ""`, this sanitizer will malform the header and break playback.

Reply:
For the first part, that makes sense.

For the second part, sounds like this is something we should fix in `recording`.

Reply:
#17842